src/components/BookForm.js

```diff
@@ -1,15 +1,30 @@
 import '../App.css';
 import React from 'react';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/books';
 
-const BookForm = () => (
-  <div>
-    <form action="/action_page.php">
-      <input type="text" name="title" value="Add your title" />
-      <br />
-      <input type="text" name="Author" value="Author's name" />
-      <input type="submit" value="Submit" />
-    </form>
-  </div>
-);
+const BookForm = () => {
+  const dispatch = useDispatch();
+
+  const handleSubmit = (e) => {
+    const bookTitle = e.target.title.value;
+    const bookAuthor = e.target.author.value;
```

Comment:
- [ ] The component should contain A local state with `bookTitle`, `bookAuthor`. They should have their inputs that update the local state every time they update (with the `onChange` handler). Since you are using a functional component you can add this state by using the `useState` hook, add a `value` property to your inputs that match your state, and add an onChange prop that sets the new value of the state.

Hint:
```
const [bookTitle, setBookTitle] = useState('')

In the form:

<input type="text" name="bookTitle" value={bookTitle} placeholder="Add your title" onChange={(e) => setBookTitle(e.target.value)} />
```